dts/bindings/mmc/zephyr,sdmmc-disk.yaml

```diff
@@ -0,0 +1,9 @@
+description: |
+  Zephyr MMC disk node. A binding with this compatible present within an SD
+  host controller device node indicates that an SDMMC disk is attached to that
+  SD bus. This binding will enable that disk to be used with the disk driver
```

Comment:
Would it make sense to introduce an MMC "bus" (`bus: mmc`) and have child devices (like this one) be on that bus?

Reply:
> Is there a `reg` that makes sense for the children of a MMC "bus". Not that is required, just asking.

Not that I can think of- SD devices will provide a "relative card address (RCA)" when requested, and selecting/deselecting cards via their RCA is how you manage multiple cards on a bus. This value is read at runtime however, and not something that is readily available for an SD device like the I2C address would be.

Reply:
> Would it make sense to introduce an MMC "bus" (`bus: mmc`) and have child devices (like this one) be on that bus?

Yes, this would make sense to do- additional work would be required to properly support multiple SD devices on a bus using this subsystem, but there's no reason not to make this binding future proof. I'll update the PR.